File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/cli/CLIEligibilityAndPermissionFragment.java
package za.co.woolworths.financial.services.android.ui.fragments.cli;

import android.animation.ObjectAnimator;
import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.core.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ScrollView;

import com.awfs.coordination.R;

import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties;
import za.co.woolworths.financial.services.android.ui.activities.CustomPopUpWindow;
import za.co.woolworths.financial.services.android.ui.views.WTextView;
import za.co.woolworths.financial.services.android.util.FragmentUtils;
import za.co.woolworths.financial.services.android.util.Utils;
import za.co.woolworths.financial.services.android.util.controller.IncreaseLimitController;

/**
 * A simple {@link Fragment} subclass.
 */
public class CLIEligibilityAndPermissionFragment extends Fragment implements View.OnClickListener {

	private static final int SLIDE_UP_ANIM_DURATION = 300;
	private LinearLayout permissionView;
	private WTextView eligibilityYes;
	private WTextView eligibilityNo;
	private WTextView permissionYes;
	private WTextView permissionNo;
	private ScrollView scrollView;
	private int paddingDp = 0;
	private LinearLayout llEligibilityView;

	public CLIEligibilityAndPermissionFragment() {
		// Required empty public constructor
	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container,
							 Bundle savedInstanceState) {
		// Inflate the layout for this fragment
		View view = inflater.inflate(R.layout.cli_eligibility_and_permission_fragment, container, false);
		permissionView = (LinearLayout) view.findViewById(R.id.permissionView);
		llEligibilityView = (LinearLayout) view.findViewById(R.id.llEligibilityView);
		eligibilityYes = (WTextView) view.findViewById(R.id.eligibilityYes);
		eligibilityNo = (WTextView) view.findViewById(R.id.eligibilityNo);
		permissionYes = (WTextView) view.findViewById(R.id.permissionYes);
		permissionNo = (WTextView) view.findViewById(R.id.permissionNo);
		scrollView = (ScrollView) view.findViewById(R.id.scrollView);
		eligibilityNo.setOnClickListener(this);
		eligibilityYes.setOnClickListener(this);
		permissionYes.setOnClickListener(this);
		permissionNo.setOnClickListener(this);

		return view;
	}

	@Override
	public void onResume() {
		super.onResume();
		Utils.setScreenName(getActivity(), FirebaseManagerAnalyticsProperties.ScreenNames.CLI_INSOLVENCY_CHECK);
	}

	@Override
	public void onClick(View view) {
		switch (view.getId()) {
			case R.id.eligibilityYes:
				eligibilityNo.setBackgroundColor(ContextCompat.getColor(getActivity(), android.R.color.transparent));
				eligibilityNo.setTextColor(ContextCompat.getColor(getActivity(), R.color.cli_yes_no_button_color));
				eligibilityYes.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.black));
				eligibilityYes.setTextColor(ContextCompat.getColor(getActivity(), R.color.white));
				permissionView.setVisibility(View.GONE);
				Utils.displayValidationMessage(CLIEligibilityAndPermissionFragment.this.getActivity(), CustomPopUpWindow.MODAL_LAYOUT.INSOLVENCY, "");
				break;
			case R.id.eligibilityNo:
				eligibilityYes.setBackgroundColor(ContextCompat.getColor(getActivity(), android.R.color.transparent));
				eligibilityYes.setTextColor(ContextCompat.getColor(getActivity(), R.color.cli_yes_no_button_color));
				eligibilityNo.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.black));
				eligibilityNo.setTextColor(ContextCompat.getColor(getActivity(), R.color.white));
				permissionView.setVisibility(View.VISIBLE);
				IncreaseLimitController ilc = new IncreaseLimitController(getActivity());
				int paddingPixel = 16;
				float density = getActivity().getResources().getDisplayMetrics().density;
				paddingDp = (int) (paddingPixel * density);
				permissionView.setPadding(0, paddingDp, 0, ilc.getScreenHeight(getActivity()));
				scrollView.post(new Runnable() {
					@Override
					public void run() {
						ObjectAnimator.ofInt(scrollView, "scrollY", permissionView.getTop()).setDuration(SLIDE_UP_ANIM_DURATION).start();
					}
				});
				Utils.setScreenName(getActivity(), FirebaseManagerAnalyticsProperties.ScreenNames.CLI_CONSENT);

				break;
			case R.id.permissionYes:
				permissionNo.setBackgroundColor(ContextCompat.getColor(getActivity(), android.R.color.transparent));
				permissionNo.setTextColor(ContextCompat.getColor(getActivity(), R.color.cli_yes_no_button_color));
				permissionYes.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.black));
				permissionYes.setTextColor(ContextCompat.getColor(getActivity(), R.color.white));
				llEligibilityView.setVisibility(View.GONE);
				permissionView.setPadding(0, paddingDp, 0, 0);
				FragmentManager fragmentManager = getActivity().getSupportFragmentManager();
				FragmentUtils fragmentUtils = new FragmentUtils();
				fragmentUtils.nextFragment(fragmentManager,
						new CLIAllStepsContainerFragment(), R.id.cliMainFrame);
				break;
			case R.id.permissionNo:
				permissionNo.setBackgroundColor(ContextCompat.getColor(getActivity(), R.color.black));
				permissionNo.setTextColor(ContextCompat.getColor(getActivity(), R.color.white));
				Utils.displayValidationMessage(CLIEligibilityAndPermissionFragment.this.getActivity(), CustomPopUpWindow.MODAL_LAYOUT.CONFIDENTIAL, "");
				break;
			default:
				break;
		}
	}
}

File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/cli/CLIEligibilityAndPermissionFragment.kt
package za.co.woolworths.financial.services.android.ui.fragments.cli

import android.animation.ObjectAnimator
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import com.awfs.coordination.R
import com.awfs.coordination.databinding.CliEligibilityAndPermissionFragmentBinding
import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties
import za.co.woolworths.financial.services.android.models.AppConfigSingleton
import za.co.woolworths.financial.services.android.models.dto.app_config.credit_limit_increase.ConfigCreditLimitIncrease
import za.co.woolworths.financial.services.android.models.dto.app_config.credit_limit_increase.ConfigEligibilityQuestions
import za.co.woolworths.financial.services.android.models.dto.app_config.credit_limit_increase.ConfigPermissions
import za.co.woolworths.financial.services.android.ui.activities.cli.CLIPhase2Activity
import za.co.woolworths.financial.services.android.ui.extension.bindColor
import za.co.woolworths.financial.services.android.ui.views.actionsheet.dialog.ErrorMessageDialog
import za.co.woolworths.financial.services.android.util.FragmentUtils
import za.co.woolworths.financial.services.android.util.KotlinUtils
import za.co.woolworths.financial.services.android.util.Utils
import za.co.woolworths.financial.services.android.util.controller.IncreaseLimitController

class CLIEligibilityAndPermissionFragment : Fragment(R.layout.cli_eligibility_and_permission_fragment), View.OnClickListener {

    companion object {
        private const val SLIDE_UP_ANIM_DURATION = 300
        private var paddingDp: Int = 0
    }

    private lateinit var binding: CliEligibilityAndPermissionFragmentBinding

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        binding = CliEligibilityAndPermissionFragmentBinding.bind(view)

        val creditLimitIncrease: ConfigCreditLimitIncrease? = AppConfigSingleton.creditLimitIncrease
        val eligibilityQuestions: ConfigEligibilityQuestions? = creditLimitIncrease?.eligibilityQuestions
        val permissions: ConfigPermissions? = creditLimitIncrease?.permissions
        (activity as? CLIPhase2Activity)?.selectedMaritalStatusPosition = null
        var eligibilityQuestionsDesc = ""
        eligibilityQuestions?.description?.forEach {
             it?.let { eligibilityQuestionsDesc += "• $it \n" }
        }

        binding.apply {
            eligibilityQuestion.text = eligibilityQuestions?.title
            eligibilityQuestionDesc?.text = eligibilityQuestionsDesc
            permissionsTitle?.text = permissions?.title
            permissionsDesc?.text = permissions?.description

            eligibilityNo?.setOnClickListener(this@CLIEligibilityAndPermissionFragment)
            eligibilityYes?.setOnClickListener(this@CLIEligibilityAndPermissionFragment)
            permissionYes?.setOnClickListener(this@CLIEligibilityAndPermissionFragment)
            permissionNo?.setOnClickListener(this@CLIEligibilityAndPermissionFragment)
        }
    }

    override fun onClick(view: View) {
        binding.apply {
            activity?.apply {
                when (view.id) {
                    R.id.eligibilityYes -> {
                        eligibilityNo?.setBackgroundColor(bindColor(android.R.color.transparent))
                        eligibilityNo?.setTextColor(bindColor(R.color.cli_yes_no_button_color))
                        eligibilityYes?.setBackgroundColor(bindColor(R.color.black))
                        eligibilityYes?.setTextColor(bindColor(R.color.white))
                        permissionView?.visibility = View.GONE
                        val infoDialog = ErrorMessageDialog(
                            R.string.cli_pop_insolvency_title,
                            R.string.cli_pop_insolvency_desc,
                            R.string.ok,
                            R.string.cli_pop_insolvency_content_desc_title,
                            R.string.cli_pop_insolvency_content_desc_desc,
                            R.string.global_button_label
                        )
                        KotlinUtils.cliErrorMessageDialog(
                            requireActivity() as? AppCompatActivity,
                            infoDialog
                        )
                    }
                    R.id.eligibilityNo -> {
                        eligibilityYes?.setBackgroundColor(bindColor(android.R.color.transparent))
                        eligibilityYes?.setTextColor(bindColor(R.color.cli_yes_no_button_color))
                        eligibilityNo?.setBackgroundColor(bindColor(R.color.black))
                        eligibilityNo?.setTextColor(bindColor(R.color.white))
                        val ilc = IncreaseLimitController(this)
                        paddingDp = 16.times(resources?.displayMetrics?.density ?: 0f).toInt()
                        permissionView?.apply {
                            visibility = View.VISIBLE
                            setPadding(0, paddingDp, 0, ilc.getScreenHeight(activity))
                            scrollView?.post {
                                ObjectAnimator.ofInt(scrollView, "scrollY", top)
                                    .setDuration(SLIDE_UP_ANIM_DURATION.toLong()).start()
                            }
                        }
                        Utils.setScreenName(
                            this,
                            FirebaseManagerAnalyticsProperties.ScreenNames.CLI_CONSENT
                        )
                    }
                    R.id.permissionYes -> {
                        permissionNo?.setBackgroundColor(
                            ContextCompat.getColor(
                                this,
                                android.R.color.transparent
                            )
                        )
                        permissionNo?.setTextColor(
                            ContextCompat.getColor(
                                this,
                                R.color.cli_yes_no_button_color
                            )
                        )
                        permissionYes?.setBackgroundColor(
                            ContextCompat.getColor(
                                this,
                                R.color.black
                            )
                        )
                        permissionYes?.setTextColor(ContextCompat.getColor(this, R.color.white))
                        llEligibilityView?.visibility = View.GONE
                        permissionView?.setPadding(0, paddingDp, 0, 0)
                        val fragmentUtils = FragmentUtils()
                        fragmentUtils.replaceFragment(
                            supportFragmentManager,
                            CLIMaritalStatusFragment.newInstance(),
                            R.id.cliMainFrame
                        )
                    }
                    R.id.permissionNo -> {
                        permissionNo?.setBackgroundColor(
                            ContextCompat.getColor(
                                this,
                                R.color.black
                            )
                        )
                        permissionNo?.setTextColor(ContextCompat.getColor(this, R.color.white))
                        val infoDialog = ErrorMessageDialog(
                            R.string.cli_pop_confidential_title,
                            R.string.cli_pop_confidential_desc,
                            R.string.ok,
                            R.string.cli_pop_confidential_content_desc_title,
                            R.string.cli_pop_confidential_content_desc_desc,
                            R.string.global_button_label
                        )
                        KotlinUtils.cliErrorMessageDialog(
                            requireActivity() as? AppCompatActivity,
                            infoDialog
                        )
                    }
                }
            }
        }
    }

    override fun onResume() {
        super.onResume()
        activity?.let { Utils.setScreenName(it, FirebaseManagerAnalyticsProperties.ScreenNames.CLI_INSOLVENCY_CHECK) }
    }

}